### PR TITLE
Refactor object types filter

### PR DESCRIPTION
**Pull request by didoda**
This provides a refactor in index pages filter for "multiple types" modules, as Objects, Media and Trash.

Object types filter is not anymore in the left side of the page: it's a checkboxes tree picker in the filter area.

Examples screenshots:

image

image

### CODE REVIEW

File: resources/js/app/components/object-types-picker/object-types-picker.vue [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    name: 'ObjectTypesPicker',
+
+    components: {
+        Treeselect,
+    },
+
+    props: {
+        initialSelected: {
+            type: Array,
+            default: () => [],
```
Comment (paolosantarsiero) [inline resources/js/app/components/object-types-picker/object-types-picker.vue:31]: Why not `default: []`?

File: resources/js/app/components/object-types-picker/object-types-picker.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        },
+        initialTypes: {
+            type: Array,
+            default: () => [],
```
Comment (paolosantarsiero) [inline resources/js/app/components/object-types-picker/object-types-picker.vue:35]: same here

File: resources/js/app/components/object-types-picker/object-types-picker.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            types: [],
+        };
+    },
+
+    mounted() {
+        this.$nextTick(() => {
+            this.types = this.initialTypes?.map((t) => ({ id: t, label: t }));
+            this.types = this.types.sort((a, b) => a.label.localeCompare(b.label));
+            this.selectedTypes = this.initialSelected.sort().map((t) => ({ id: t, label: t }));
+            console.log(this.selectedTypes);
```
Comment (paolosantarsiero) [inline resources/js/app/components/object-types-picker/object-types-picker.vue]: you can remove `console.log`

File: resources/js/app/components/object-types-picker/object-types-picker.vue [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        });
+    },
+
+    methods: {
+        changeTypes() {
+            const types = [];
+            for (const item of this.selectedTypes) {
+                types.push(item.id);
+            }
```
Comment (paolosantarsiero) [inline resources/js/app/components/object-types-picker/object-types-picker.vue]: you can use `const typesId = this.selectedTypes.map((item) => item.id);`

File: resources/js/app/components/object-types-picker/object-types-picker.vue (hunk 2 above)
Comment (didoda) [reply resources/js/app/components/object-types-picker/object-types-picker.vue:35]: It's a little verbose, but better in ES6

File: resources/js/app/components/object-types-picker/object-types-picker.vue (hunk 1 above)
Comment (didoda) [reply resources/js/app/components/object-types-picker/object-types-picker.vue:31]: It's a little verbose, but better in ES6

File: resources/js/app/components/object-types-picker/object-types-picker.vue [hunk 5]
```diff
@@ -48,7 +48,7 @@ export default {
         this.$nextTick(() => {
             this.types = this.initialTypes?.map((t) => ({ id: t, label: t }));
             this.types = this.types.sort((a, b) => a.label.localeCompare(b.label));
-            this.selectedTypes = this.initialSelected.sort().map((t) => ({ id: t, label: t }));
+            this.selectedTypes = this.initialSelected ? this.initialSelected.sort().map((t) => ({ id: t, label: t })) : [];
```
Comment (paolosantarsiero) [inline resources/js/app/components/object-types-picker/object-types-picker.vue]: `this.selectedTypes = (this.initialSelected ?? []).sort().map((t) => ({ id: t, label: t }))`

**Pull request merged**